src/amo/components/LanguagePicker.js

```diff
@@ -1,14 +1,59 @@
+/* global window */
+import config from 'config';
 import React, { PropTypes } from 'react';
 import { compose } from 'redux';
+import { connect } from 'react-redux';
 
+import { languageJSON } from 'core/languages';
 import translate from 'core/i18n/translate';
+import { addQueryParams } from 'core/utils';
 
 import './LanguagePicker.scss';
 
 
+export function changeLocaleURL({ currentLocale, location, newLocale }) {
+  const newPath = location.pathname.replace(`/${currentLocale}/`,
+    `/${newLocale}/`);
+  return addQueryParams(newPath, location.query);
+}
+
 export class LanguagePickerBase extends React.Component {
   static propTypes = {
+    currentLocale: PropTypes.string.isRequired,
     i18n: PropTypes.object.isRequired,
+    location: PropTypes.object.isRequired,
+    _window: PropTypes.object,
+  }
+
+  changeLanguage = (event) => {
+    event.preventDefault();
+
+    const { currentLocale, location, _window } = this.props;
+    const newLocale = event.target.value;
+
+    if (currentLocale !== newLocale) {
+      const newURL = changeLocaleURL({ currentLocale, location, newLocale });
+      // We change location because a locale change requires a full page
+      // reload to get the new translations, etc.
+      (_window || window).location = newURL;
+    }
+  }
+
+  languages() {
+    const { currentLocale } = this.props;
+
+    return config.get('langs').map((locale) => {
+      if (languageJSON[locale]) {
+        const isCurrentLocale = currentLocale === locale ? 'selected' : '';
```

Comment:
Nice! I noticed that in the console output too after you pointed this out, oops 🙈